client/driver/docker.go

```diff
@@ -1028,3 +1030,13 @@ func (h *DockerHandle) collectStats() {
 		}
 	}
 }
+
+func calculatePercent(newSample, oldSample, newTotal, oldTotal uint64, cores int) float64 {
+	numerator := newSample - oldSample
+	denom := newTotal - oldTotal
+	if numerator <= 0 || denom <= 0 {
+		return 0.0
+	}
+
+	return (float64(numerator) / float64(denom)) * float64(cores) * 100.0
```

Comment:
Can you guard against `cores` being zero?  There isn't a guarantee that `s.CPUStats.CPUUsage.PercpuUsage` is non-empty.